clinvar/src/main/scala/org/broadinstitute/monster/etl/clinvar/models/intermediate/VariationArchive.scala

```diff
@@ -229,16 +230,12 @@ object VariationArchive {
         rawScv =>
           val submitter = Submitter.fromRawAssertion(rawScv)
           val submission = Submission.fromRawAssertion(submitter, rawScv)
-          val scv = SCV.fromRawAssertion(
-            variation,
-            vcv,
-            submitter,
-            submission,
-            rawScv
-          )
-
-          // Extract variation-related data from the SCV.
-          scvVariations.appendAll(SCVVariation.allFromRawAssertion(scv, rawScv))
+          val scvAccessionId = rawScv
+            .extract("ClinVarAccession", "@Accession")
+            .getOrElse {
+              throw new IllegalStateException(s"Found an SCV with no ID: $rawScv")
+            }
+            .str
```

Comment:
I'd rather we keep detailed logic like this encapsulated within the model's companion object. I can think of two approaches:
1. Have `SCV.fromRawAssertion` return an `SCV` with `Array.empty` for its trait set, use that partial object to build the SCV trait-sets and traits, then use `.copy` to make a new `SCV` with the trait sets included.
2. Make a new `SCV.extractId` method containing this logic and call it here to get the ID. Keep the rest of the PR here the same.

1 feels like cleaner separation of responsibility, but the code will be more complex.